crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/prosecution_concluded/controller/ProsecutionConcludedController.java

```diff
@@ -4,27 +4,28 @@
 import io.swagger.v3.oas.annotations.tags.Tag;
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
-import org.springframework.http.HttpHeaders;
 import org.springframework.http.MediaType;
 import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.*;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
 import uk.gov.justice.laa.crime.crowncourt.prosecution_concluded.service.ProsecutionConcludedDataService;
 
 @Slf4j
 @RestController
 @RequiredArgsConstructor
 @RequestMapping("api/internal/v1/proceedings/prosecution-concluded/{maatId}/messages")
-@Tag(name = "Prosecution concluded", description = "Rest API for Prosecution concluded")
+@Tag(name = "Prosecution concluded", description = "Rest API for prosecution concluded messages")
 public class ProsecutionConcludedController {
 
-    private final ProsecutionConcludedDataService prosecutionConcludedDataService;
+    private final ProsecutionConcludedDataService concludedDataService;
 
     @GetMapping(value = "/count", produces = MediaType.APPLICATION_JSON_VALUE)
     @Operation(description = "Retrieve Prosecution Concluded Count")
     public ResponseEntity<Object> getCountByMaatIdAndStatus(@PathVariable int maatId,
-                                                            @RequestParam(value = "status", defaultValue = "PENDING") String  status) {
-        HttpHeaders responseHeaders = new HttpHeaders();
-        responseHeaders.setContentLength(prosecutionConcludedDataService.getCountByMaatIdAndStatus(maatId, status));
-        return ResponseEntity.ok().headers(responseHeaders).build();
+            @RequestParam(value = "status", defaultValue = "PENDING") String status) {
+        return ResponseEntity.ok(concludedDataService.getCountByMaatIdAndStatus(maatId, status));
```

Comment:
I do believe that the JSON return value from this response, given that it's a string of a count, is not valid JSON (if it is of something like `{"12"}` etc. Something to perhaps consider

Reply:
It is indeed valid JSON, in future we may want something like {"count": "2"}, but I decided not to go down that route in this ticket.

Reply:
I've double-checked to make sure I'm not going crazy but I don't think it is valid JSON. I tried to run it through some validation in IntelliJ and it won't even compile because it detects the string isn't valid, expecting a `:` where it receives a `}` here:

![Screenshot 2025-04-01 at 11 38 00](https://github.com/user-attachments/assets/a2f02197-a101-4616-946b-ad5ea3a613d9)

I've run it through an online validator too and receive a similar output that the JSON isn't valid.

Reply:
Thanks for sending through the RFC, very helpful 👍